arviz/plots/kdeplot.py

```diff
@@ -146,6 +147,10 @@ def plot_kde(
 
     figsize, *_, xt_labelsize, linewidth, markersize = _scale_fig_size(figsize, textsize, 1, 1)
 
+    if isinstance(values, xr.Dataset):
+        raise ValueError("Xarray dataset object detected. Use plot_posterior instead of plot_kde")
```

Comment:
I  would suggest the user to use plot_posterior, plot_density, plot_joint or plot_joint.